### PR TITLE
Clover bead: componentize ProjectAvatar

### PR DESCRIPTION
**Pull request by modernserf**
## Links
* Remix link: https://clover-bead.glitch.me

## Changes:
* componentize ProjectAvatar

## How To Test:
* visit https://clover-bead.glitch.me
* check the new project menu
* check the search autocomplete results
* check any of the popover menus that have a project search (e.g. add project to team, add project to collection)

### CODE REVIEW

File: src/components/images/avatar.js [hunk 1]
```diff
@@ -95,3 +96,26 @@ UserAvatar.defaultProps = {
   hideTooltip: false,
   withinButton: false,
 };
+
+export const ProjectAvatar = ({ project, withinButton }) => (
```
Comment (clottman) [inline src/components/images/avatar.js]: I think for accessibility, we need to be setting alt where the project avatar is being displayed to convey info about the project (name is only available as a tooltip) and setting alt='' where the name is displayed right next to the avatar.

Comment (clottman) [reply src/components/images/avatar.js]: Also places where the project avatar is used as a button should have alt text, but if ProjectAvatar is a button within something that's already a button leading to the project page, we should make it not a button (like in search results for a project)

File: src/components/team-analytics/team-analytics-project-details.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ const RECENT_REMIXES_COUNT = 100;
 const ProjectDetails = ({ projectDetails }) => (
   <article className={styles.projectDetails}>
     <ProjectLink project={projectDetails}>
-      <ProjectAvatar {...projectDetails} />
+      <ProjectAvatar project={projectDetails} hasAlt />
```
Comment (modernserf) [inline src/components/team-analytics/team-analytics-project-details.js]: @clottman this one _should_ have the alt, right? since it's by itself in a link?

Comment (clottman) [reply src/components/team-analytics/team-analytics-project-details.js]: Yep exactly!

**Pull request merged**